packages/std/src/imports.rs

```diff
@@ -196,14 +196,17 @@ impl Api for ExternalApi {
         let pubkey_send_ptr = &*pubkey_send as *const Region as u32;
 
         let result = unsafe { verify_secp256k1(hash_send_ptr, sig_send_ptr, pubkey_send_ptr) };
-        if result != 0 {
-            let error = unsafe { consume_string_region_written_by_vm(result as *mut Region) };
-            return Err(StdError::generic_err(format!(
-                "secp256k1_verify error: {}",
-                error
-            )));
+        match result {
+            1 => Ok(true),
+            0 => Ok(false),
+            ptr => {
+                let error = unsafe { consume_string_region_written_by_vm(ptr as *mut Region) };
+                return Err(StdError::generic_err(format!(
+                    "verify_secp256k1 error: {}",
+                    error
+                )));
+            }
```

Comment:
No need to handle the `ptr` case anymore since we never `write_to_contract` in the VM (which is good)

Reply:
Will leave this as it is then, just for closure of the `match` options.

Reply:
Let's just panic in this case. It means the VM is writing garbage to the contract, which we do not want to process. We also do not want to increase code size for handling impossible errors.